app/scripts/services/validateRegistrant.js

```diff
@@ -68,7 +71,7 @@ angular.module('confRegistrationWebApp')
       var blocks = page ? _.find(conference.registrationPages, {id: page}).blocks : _.flatten(conference.registrationPages, 'blocks');
 
       angular.forEach(blocks, function(block){
-        if (!block.required || block.adminOnly || !blockVisibleRuleCheck(block, registrant) || !blockInRegistrantType(block, registrant)) { return; }
+        if (!block.required || block.adminOnly || !blockVisibleRuleCheck(block, registrant, ruleTypeConstants.SHOW_QUESTION) || !blockInRegistrantType(block, registrant)) { return; }
```

Comment:
Javascript isn't very strict. Anything you don't pass results in undefined (which is similar but very different to `null`). If you want checking like that I think you'd need to migrate to Typescript (which is recommended for Angular 2) or something similar.

I think this is one of my favorite images:
![Javascript the good parts](http://www.fvi.edu/wp-content/uploads/2016/09/wR3ZxfB.jpg)